iceoryx_hoofs/test/moduletests/test_cxx_function_ref.cpp

```diff
@@ -389,8 +322,7 @@ TEST_F(function_refTest, CreationWithFunctionNullPointerIsNotCallable)
     ::testing::Test::RecordProperty("TEST_ID", "e10c9dd4-d8a2-4231-a6c0-0e93e65e1ae0");
     int (*fp)(int) = nullptr;
     function_ref<int(int)> sut(fp);
-
-    EXPECT_FALSE(sut.operator bool());
+    EXPECT_DEATH(sut(42), "Empty function_ref invoked");
```

Comment:
Here, I would prefer to remove this c'tor as `nullptr` can be passed and hence it's possible to create an invalid `function_ref`. @MatthiasKillat I can' recall the reason why you need this c'tor. Why is it needed?

Reply:
@mossmaurice I would not remove the ctor I would change the argument from a function pointer to a function reference `function_ref(ReturnType (&function)(ArgTypes...)) noexcept` then we should be safe and still support c functions in function refs.
Maybe we are then not as generic as the STL is but in my opinion this is completely reasonable since we have a safety use case. And with a lambda one can always solve this issue

Reply:
Yes, make it unable to reference the null-function this way. The ctor was needed because constructing a `function_ref` from a function pointer it did not quite work correctly with universal references.